### PR TITLE
search filters: refactoring to be more generic

### PR DESCRIPTION
**Pull request by Garfield-fr**
The mocklab service no longer exists. Its data have been migrated
to the new wiremock service.

* Fixes api mock for the tester.
* Changes search filters processing and display.
* Updates search filters on the document route.

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/record.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ export interface SearchFilter {
   filter: string;
   label: string;
   value: string;
+  showIfQuery?: boolean;
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/record.ts:71]: Can you add some comment to explain the use of this new option?

File: projects/rero/ng-core/src/lib/record/record.ts [hunk 2]
```diff
@@ -81,6 +82,11 @@ export interface SearchFilter {
   };
 }
 
+export interface SearchFilterSection {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/record.ts:85]: Can you add a comment?

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 3]
```diff
@@ -1089,10 +1105,36 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
...
+
+  /**
+   * Show the filter
+   * @param searchFilter - search Filter
+   * @returns true if the filter is show
+   */
+  showFilter(searchFilter: SearchFilter) {
+    if (this._config.allowEmptySearch) {
+      return true;
+    } else if (!this.q) {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:1128]: else are not required has we return the value at each step.

File: proxies/wiremockapi.json [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
   },
   "/schemas/*": {
-    "target": "https://l985g.mocklab.io",
+    "target": "https://l985g.wiremockapi.cloud",
     "secure": false,
     "logLevel": "debug",
     "changeOrigin": true,
     "pathRewrite": {
-      "^/schemas": "https://l985g.mocklab.io/schemas"
+      "^/schemas": "https://l985g.wiremockapi.cloud/schemas"
```
Comment (jma) [inline proxies/wiremockapi.json:17]: Is the service API changed? Can you mention this in the commit message?

**Review by jma — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

**Pull request merged**